org.eclipse.xtext/src/org/eclipse/xtext/service/AllRulesCache.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) 2018 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.xtext.service;
+
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.emf.common.notify.Adapter;
+import org.eclipse.emf.common.notify.Notifier;
+import org.eclipse.emf.common.notify.impl.AdapterImpl;
+import org.eclipse.xtext.AbstractRule;
+import org.eclipse.xtext.Grammar;
+import org.eclipse.xtext.GrammarUtil;
+import org.eclipse.xtext.TerminalRule;
+
+import com.google.common.collect.Maps;
+
+/**
+ * @author Christian Dietrich - Initial contribution and API
+ * @since 2.14
+ */
+public class AllRulesCache {
+	public static class AllRulesCacheAdapter extends AdapterImpl {
+		private final AllRulesCache element;
+
+		public AllRulesCacheAdapter(final AllRulesCache element) {
+			this.element = element;
+		}
+
+		public AllRulesCache get() {
+			return element;
+		}
+
+		@Override
+		public boolean isAdapterForType(final Object object) {
+			return object == AllRulesCache.class;
+		}
+	}
+
+	private final Map<String, AbstractRule> allRules;
+
+	public AllRulesCache(final Grammar grammar) {
+		allRules = Maps.newHashMap();
+		List<AbstractRule> allRules = GrammarUtil.allRules(grammar);
+		for (final AbstractRule rule : allRules) {
+			if ((rule instanceof TerminalRule)) {
```

Comment:
Unnecessary parentheses. Please use code cleanup actions on the ported code.

Reply:
cleanup actions do not cover that :( I've already performed them